### PR TITLE
Fix issue #29837: added test case for aggregation with isnan

### LINKED ISSUE #29837: groupby() drops categorical columns when aggregating with isna()
#### Code Sample, a copy-pastable example if possible

```python
df = pd.DataFrame({'A': [1, 1, 1, 1],
                   'B': [1, 2, 1, 2],
                   'numerical_col': [.1, .2, np.nan, .3],
                   'object_col': ['foo','bar','foo','fee'],
                   'categorical_col': ['foo','bar','foo','fee']
                  })

df = df.astype({'categorical_col':'category'})

df.groupby(['A','B']).agg(lambda df: df.isna().sum())

#		numerical_col	object_col
# A	B		
# 1	1	1.0                   0
#       2	0.0	              0

```
#### Problem description

The categorical column "categorical_col" is expected to survive the aggregation, however, it gets dropped.

#### Expected Output
``` python
#		numerical_col	object_col categorical_col
# A	B		
# 1	1	1.0                   0                 0
#       2	0.0	              0                 0
```
#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.1.final.0
python-bits: 64
OS: Linux
OS-release: 3.10.0-693.11.6.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: 4.3.1
pip: 19.3.1
setuptools: 40.6.3
Cython: None
numpy: 1.15.4
scipy: 1.1.0
pyarrow: 0.11.1
xarray: None
IPython: 7.1.1
sphinx: None
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: 2.6.1
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.5
lxml: 4.3.0
bs4: None
html5lib: None
sqlalchemy: 1.2.13
pymysql: 0.9.3
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by biddwan09**
- [x] closes #29837
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@biddwan09 thanks for the PR. FYI Linux py37_np_dev test failures unrelated, xref #35041

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 1]
```diff
@@ -994,3 +994,30 @@ def test_groupby_get_by_index():
...
+    index = pd.MultiIndex.from_arrays([[1, 1], [1, 2]], names=("A", "B"))
+    expected = pd.DataFrame(
+        data={
+            "numerical_col": [1.0, 0.0],
+            "object_col": [0, 0],
+            "categorical_col": [0, 0],
+        },
+        index=index,
+    )
+    pd.testing.assert_frame_equal(result, expected)
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1023]:
```suggestion
    tm.assert_frame_equal(result, expected)
```

can you also change the previous test to use tm.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @biddwan09 generally lgtm.

**Comment by biddwan09** [comment]
Hi @simonjayhawkins let me know if any other change is required

**Comment by simonjayhawkins** [comment]
Thanks @biddwan09

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 2]
```diff
@@ -994,3 +994,30 @@ def test_groupby_get_by_index():
     res = df.groupby("A").agg({"B": lambda x: x.get(x.index[-1])})
     expected = pd.DataFrame(dict(A=["S", "W"], B=[1.0, 2.0])).set_index("A")
     pd.testing.assert_frame_equal(res, expected)
+
+
+def test_aggregate_categorical_with_isnan():
+    # GH 29837
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1000]: prob should have been in test_category.py

Comment (simonjayhawkins) [reply pandas/tests/groupby/aggregate/test_aggregate.py:1000]: There are two other tests for categorical in that file, test_agg_cython_category_not_implemented_fallback and test_aggregate_categorical_lost_index. move those as well?

Comment (jreback) [reply pandas/tests/groupby/aggregate/test_aggregate.py:1000]: yeah prob

Comment (simonjayhawkins) [reply pandas/tests/groupby/aggregate/test_aggregate.py:1000]: moved in #35052

**Pull request merged**